### ISSUE TITLE
Unable to load actor document, html returned for any Accept: request

### ISSUE THREAD

**Issue by redmatrix** [Plume-org/Plume#106]
I've made a request for the ActivityPub actor document (discovered in one of the current issues)

https://baptiste.gelez.xyz/@/kaniini/

Using this HTTP Accept header:

Accept: application/ld+json; profile="https://www.w3.org/ns/activitystreams", application/ld+json;profile="https://www.w3.org/ns/activitystreams", application/activity+json, application/ld+json

This header works for every AP/ActivityStreams installation I've tested against previously. The second variant is for sites that don't have a space between the mimetype and the profile; which I've found can be problematic.  

I've tried several variations and in all cases a text/html document is returned, even though that is not permitted by the Accept header. (Should return 406 if I recall correctly). But what I would really like to see returned is an ActivityPub actor document and not certain how to do that given that the Accept header isn't being honoured. It isn't application specific (I'm using hubzilla and red); but also occurs if one uses this Accept header in a browser using the browser developer tools. I've also added application/json to the header and see nothing different. I've also tried without the trailing slash in the URL. 

How can I obtain an ActivityPub actor document?

**Comment by fdb-hiroshima** [comment]
We are matching against `application/activity+json`, however it seems it need to be the first type proposed for it to work (also specifying no Accept header load the actor, but I don't think it can be relied on in the future). We should probably search for a way to work with such Accept header containing more than one content type.

**Comment by BaptisteGelez** [comment]
Matching the AP route when there is more than one content type is probably a Rocket issue, as it does all the routing.

**Comment by redmatrix** [comment]
Thanks. application/activity+json is legal for ActivityStreams but not ActivityPub, which required an Accept type of 

````
application/ld+json; profile="https://www.w3.org/ns/activitystreams"
````

last time I looked (7-8 months ago). This isn't a show-stopper today but may be an issue going forward, as I'm working on a suite of products/projects which are ActivityStreams compliant but *not* ActivityPub compliant and will provide completely different content if you choose the wrong one first.

**Comment by BaptisteGelez** [comment]
Fixed in b95e384
